packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import { SafeCall } from "src/libraries/SafeCall.sol";
+import { CrossL2Inbox } from "src/L2/CrossL2Inbox.sol";
+import { Encoding } from "src/libraries/Encoding.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @custom:proxied
+/// @custom:predeploy 0x4200000000000000000000000000000000000023
+/// @title L2ToL2CrossDomainMessenger
+contract L2ToL2CrossDomainMessenger is ISemver {
+    /// @notice Emitted whenever a message is successfully relayed on this chain.
+    /// @param msgHash Hash of the message that was relayed.
+    event RelayedMessage(bytes32 indexed msgHash);
+
+    /// @notice Emitted whenever a message fails to be relayed on this chain.
+    /// @param msgHash Hash of the message that failed to be relayed.
+    event FailedRelayedMessage(bytes32 indexed msgHash);
+
+    /// @notice Slot for the sender of the current cross domain message.
+    /// @dev Equal to bytes32(uint256(keccak256("l2tol2crossdomainmessenger.sender")) - 1)
+    bytes32 public constant CROSS_DOMAIN_MESSAGE_SENDER_SLOT =
+        0xb83444d07072b122e2e72a669ce32857d892345c19856f4e7142d06a167ab3f3;
+
+    /// @notice Current message version identifier.
+    uint16 public constant MESSAGE_VERSION = uint16(0);
+
+    /// @notice Initial balance for the contract.
+    uint248 public constant INITIAL_BALANCE = type(uint248).max;
+
+    /// @notice Address of the L2 Cross Domain Messenger on this chain.
+    address public crossL2Inbox;
```

Comment:
This is never set to anything, we don't need this

Reply:
are you sure we don't need it? `crossL2Inbox` is used in the `relayMessage` method to check that the sender is `CrossL2Inbox` and that the origin in `CrossL2Inbox` is this contract. the value for `crossL2Inbox` is set through l2 genesis (which is why in the contract is not set to anything. maybe we could add a constructor to `L2ToL2CrossDomainMessenger` that additionally sets it to `Predeploys.CROSS_L2_INBOX`? this would be unnecessary though, since the value is already set